### DISCUSSION TITLE
eslint-plugin-sonar/sonarjs

### DISCUSSION THREAD

**Discussion by Goldziher** [oxc-project/oxc#4863] [Feature Request]
SonarQube has a very useful EsLint plugin (eslint-plugin-sonarjs) - I was always a fan of their "cognitive complexity" over other implementations of complexity measurement for example, and their rules are very useful as a style guide as well. Recently a new plugin was released that extracts all the rules from Sonar (eslint-plugin-sonar), which isn't official. 

It would be great to have at least some of the rules from these plugins implemented in oxclint. 

For reference:

[eslint-plugin-sonarjs](https://github.com/SonarSource/eslint-plugin-sonarjs)
[eslint-plugin-sonar](https://www.npmjs.com/package/eslint-plugin-sonar)

**Comment by Goldziher** [comment]
@rzvxa @DonIsaac whats the rfc process?

**Comment by DonIsaac** [comment]
I was thinking about this too, but we may run into problems considering it's LGPL licensed.

**Reply by Goldziher** [reply]
The LGPL license is for their code, it can't cover the functionality itself.

**Reply by DonIsaac** [reply]
We'd then need to guarantee that every contribution was created without reading their code... which isn't really viable if we're taking public contributions. 

We could potentially port over some rules, but only if maintainers wrote them and were very careful about how they did so. I'm not sure this is worth it in most cases, but it _is_ very tempting for some of their rules.

**Reply by Boshen** [reply]
Biome implemented one too https://biomejs.dev/linter/rules/no-excessive-cognitive-complexity/

**Reply by Goldziher** [reply]
Or it's possible to ask for permission?

**Comment by Hideman85** [comment]
Hi, jumping in this discussion, I also beleive there is quite some great rules that would be very nice to be seen ported to oxlint. I'm already enjoying a lot the port of unicorn rules and I'm thinking sonar rules is a great addition.

Furthermore, in corporate environment, sonar is well known and used for the vulnerabilities and quality scanning over the codebase. Having oxlint blazing fast checking those rules and auto-fixing some of them is a great value.

Looking forward to see if this is something that could be integrated or not.

**Comment by alfaproject** [comment]
We also require sonarjs rules to work before we can switch to Oxlint and we won't keep both ESLint and Oxlint in the same codebase, so until Oxlint supports sonarjs I guess we are out of luck.

The problem with sonarjs rules is that some of them are type-aware and I don't think that's supported by Oxlint JS plugins implementation atm

**Reply by camc314** [reply]
You can use it as a custom plugin. 


Please track https://github.com/oxc-project/oxc/issues/19596 for type access for custom plugins

**Reply by alfaproject** [reply]
Oh nice, that's good news, thanks!
